tests/telemetry/test_telemetry.py

```diff
@@ -12,6 +12,31 @@
 from conftest import _write_sample_conda_env, _prepare_files
 
 
+# The below fixtures are to mock the different virtual environments
+# Ref: https://stackoverflow.com/questions/51266880/detect-if-
+# python-is-running-in-a-conda-environment
+@pytest.fixture()
+def inside_conda_env(monkeypatch):
+    monkeypatch.setenv('CONDA_PREFIX', True)
+
+
+# Ref: https://stackoverflow.com/questions/1871549/
+# determine-if-python-is-running-inside-virtualenv
+@pytest.fixture()
+def inside_pip_env(monkeypatch):
+    monkeypatch.setattr(telemetry.sys, 'prefix', 'sys.prefix')
+    monkeypatch.setattr(telemetry.sys, 'base_prefix', 'base_prefix')
+
+
+# Ref: https://stackoverflow.com/questions/43878953/how-does-one-detect-if-
+# one-is-running-within-a-docker-container-within-python
+@pytest.fixture()
+def inside_docker(monkeypatch, tmp_directory):
+    mock = Mock()
+    mock.return_value = True
+    monkeypatch.setattr(telemetry.os.path, 'exists', mock)
```

Comment:
this fixture will return true for all calls to .exists(), but you only want a subset of those. check out Mocks documentation, there is a way to define an output when the function is called with certain inputs. e.g. return True if .exists is called with '/.dockerenv' but false with all other values

Reply:
Wow this was a hard one. Not necessarily worth it since it's isolated in the test.